src/bridger/Bridger.sol

```diff
@@ -0,0 +1,201 @@
+    using MessageHashUtils for bytes32;
+    using SignatureChecker for address;
+
+    /* ============ Events ============ */
+    event Deposit(address indexed from, address indexed asset, uint256 amount, address indexed assetBought, uint256 amountBought);
+
+    /* ============ Constants ============ */
+    address public constant REFUND_L2_ACCOUNT = address(1);
+    address public constant SENDER_ACCOUNT = address(1);
+
+    /* ============ State Variables ============ */
+    address public immutable override arbitrumL1GatewayRouter;
+    /// @dev Mapping of all depositors by user address and asset address
+    mapping(address => mapping(address => uint256)) public override deposits;
+    /// @dev We include a nonce in every hashed message, and increment the nonce as part of a
+    /// state-changing operation, so as to prevent replay attacks, i.e. the reuse of a signature.
+    mapping(address => uint256) public override nonces;
+    /// @dev Count of deposits
+    uint256 public depositCount;
+
+    /* ============ Constructor & Upgrades ============ */
+    constructor(address _arbitrumL1GatewayRouter) {
+        _disableInitializers();
+        arbitrumL1GatewayRouter = _arbitrumL1GatewayRouter; // 0xD9041DeCaDcBA88844b373e7053B4AC7A3390D60
+    }
+
+    /**
+     * @dev Upgrade calling `upgradeTo()`
+     */
+    function initialize() external initializer {
+        __Ownable_init(msg.sender);
+        __UUPSUpgradeable_init();
+    }
+
+    /**
+     * @dev Authorize the upgrade. Only by an owner.
+     * @param newImplementation address of the new implementation
+     */
+    // This function is called by the proxy contract when the factory is upgraded
+    function _authorizeUpgrade(address newImplementation) internal view override {
+        (newImplementation);
+        if (msg.sender != owner()) revert OnlyOwner();
+    }
+
+    /* ============ Deposit methods ============ */
+
+    /**
+     * @dev Deposit the specified amount of tokens in to the Kinto L2
+     * @param _kintoWallet Kinto Wallet Address on L2 where tokens will be deposited
+     * @param _signatureData Struct with all the required information to deposit via signature
+     * @param _swapData Struct with all the required information to swap the tokens
+     * @param _permitSignature Signature to be recovered to allow the spender to spend the tokens
+     */
+    function depositBySig(
+        address _kintoWallet,
+        IBridger.SignatureData calldata _signatureData,
+        IBridger.SwapData calldata _swapData,
+        bytes calldata _permitSignature
+    ) external override onlySignerVerified(_signatureData) {
+        if (msg.sender != owner() && msg.sender != SENDER_ACCOUNT) {
+            revert OnlySender();
+        }
+        _permit(_signatureData.inputAsset, _signatureData.amount, _signatureData.expiresAt, _permitSignature);
+        // swap using 0x
+        uint256 amountBought = _fillQuote(
```

Comment:
if i'm not mistaken, after the `_fillQuote` is executed you may have:
- a remanent of tokens (imagine you want to swap USDC -> STETH, it might be that not all of your USDC got swapped so we need to refund that back to the user)
- a remanent of the gas (ETH) that we should also refund to the user

Reply:
We want to keep all that because we are paying the gas

Reply:
but shouldn't the token remanents go back to the user? and if we pay the gas fees, the contract will keep on accumulating these ETH remanents so ideally we should have a way to withdraw them